ocaml/lib/util.ml

```diff
@@ -1,6 +1,25 @@
 open Core_kernel
 module Js = Js_of_ocaml.Js
 
+external get_ts_bindings : unit -> Js.Unsafe.any Js.Optdef.t = "getTsBindings"
+
+(* the ?. operator from JS *)
+let ( |. ) (value : _ Js.Optdef.t) (key : string) =
+  Js.(
+    if phys_equal value undefined then undefined
+    else Unsafe.get value (string key))
+
+module Js_environment = struct
+  type t = Node | Web | Unknown
+
+  let value =
+    let env = get_ts_bindings () |. "jsEnvironment" in
+    Js.(
+      if phys_equal env (def (string "node")) then Node
+      else if phys_equal env (def (string "web")) then Web
+      else Unknown)
+end
```

Comment:
this makes the ocaml bindings aware of the current JS environment. I originally thought that I needed this to avoid using nodejs filesystem operations on the web. in the design I ended up with, the `write` and `read` operations are generic and can be set to something which works in either nodejs or the web on the JS side; so this remains unused.
It still seems useful to have.